### DISCUSSION TITLE
Performance impact

### DISCUSSION THREAD

**Discussion by matejdro** [Anamorphosee/stacktrace-decoroutinator#12] [General]
Hello,

looking at this project, it looks very promising. However, official stacktrace recovery method is known to cause a pretty big performance hit and thus is not suitable to be ran in production.

What is the performance hit of this method?

**Comment by Anamorphosee** [comment]
For JVM, [according to my measurement](https://github.com/Anamorphosee/stacktrace-decoroutinator/blob/master/stacktrace-decoroutinator-jvm/src/test/kotlin/performance-test-jvm.kt), the performance overhead for 1 coroutine "awaking"  is less than 1 ms even for huge stack trace (depth is 500 calls). More common stack trace depth (10 - 50 calls) "awakes" in about 0.1 - 0.3 ms.

My measurement method is not perfect, but still can give a representation of the performance hit.
